src/eachkmer.jl

```diff
@@ -75,27 +75,27 @@ end
     k = kmersize(T)
     pos, kmer = state
     pos += it.step
-    has_ambiguous = false
+    ambig_or_gap = false
 
     # faster path: return the next overlapping kmer if possible
     if it.step < k && pos + k - 1 ≤ endof(it.seq)
         offset = k - it.step
         if it.step == 1
             nt = inbounds_getindex(it.seq, pos+offset)
             kmer = kmer << 2 | trailing_zeros(nt)
-            has_ambiguous |= isambiguous(nt)
+            ambig_or_gap |= (isambiguous(nt) || isgap(nt))
```

Comment:
Good catch! I should have seen `iscertain` was applicable here, I used it instead of isambiguous and isgap in the other site counting stuff I've done. I'll make the iscertain change now, and add a test case in the morning - it's too late here!